core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+
+        title_limit = constants.MAX_CHARS_IN_BLOG_POST_TITLE
+        if len(title) > title_limit:
+            raise utils.ValidationError(
+                'blog post title should at most have %d chars, received: %s'
+                % (title_limit, title))
+
+        if strict:
+            if len(title) == 0:
+                raise utils.ValidationError('Title should not be empty')
+
+    @classmethod
+    def require_valid_tags(cls, tags, strict):
+        """Validates tags for the blog post object.
+
+        Args:
+            tags: list(str). The list of tags assigned to a blog post.
+            strict: bool. Enable strict checks on the blog post when the blog
+                post is published or is going to be published.
+
+        Raises:
+            ValidationErrors.
+        """
+        if not isinstance(tags, list):
+            raise utils.ValidationError(
+                'Expected \'tags\' to be a list, received: %s' % tags)
+        for tag in tags:
+            if not isinstance(tag, python_utils.BASESTRING):
+                raise utils.ValidationError(
+                    'Expected each tag in \'tags\' to be a string, received: '
+                    '\'%s\'' % tag)
+
+            if len(tag) == 0:
+                raise utils.ValidationError(
+                    'Tag should not be empty, received: \'\'')
+
+            if not re.match(constants.TAG_REGEX, tag):
+                raise utils.ValidationError(
+                    'Tags should only contain lowercase letters and spaces, '
+                    'received: \'%s\'' % tag)
+
+            if (tag[0] not in string.ascii_lowercase or
+                    tag[-1] not in string.ascii_lowercase):
+                raise utils.ValidationError(
+                    'Tags should not start or end with whitespace, received: '
+                    '\'%s\'' % tag)
+
+            if re.search(r'\s\s+', tag):
+                raise utils.ValidationError(
+                    'Adjacent whitespace in tags should be collapsed, '
+                    'received: \'%s\'' % tag)
+
+        if strict:
+            if len(tags) == 0:
+                raise utils.ValidationError(
+                    'Atleast one tag should be selected')
+
+        if len(set(tags)) != len(tags):
+            raise utils.ValidationError('Some tags duplicate each other')
+
+    def to_dict(self):
+        """Returns a dict representing this blog post summary domain object.
+
+        Returns:
+            dict. A dict, mapping all fields of blog post instance.
+        """
+        return {
+            'id': self.id,
+            'author_name': user_services.get_username(
+                self.author_id),
+            'title': self.title,
+            'summary': self.summary,
+            'thumbnail_filename': self.thumbnail_filename,
+            'tags': self.tags,
+            'url_fragment': self.url_fragment,
+        }
+
+    def serialize(self):
```

Comment:
Do we also need deserialize function?

Reply:
As per offline discussion, we found we only need these function for cacheable models and the blog post models are not expected to be cached so we are going to remove the serialize and deserialize functions*

Reply:
Removed serialized.